lib/features/auth/data/rest/model/user_json.dart

```diff
@@ -0,0 +1,18 @@
+import 'package:flutter_demo/core/domain/model/user.dart';
+
+class UserJson {
+  UserJson({required this.id, required this.username});
+
+  factory UserJson.fromJson(Map<String, dynamic> json) => UserJson(
+        id: json["id"] as String? ?? '',
+        username: json["username"] as String? ?? '',
+      );
+
+  final String? id;
+  final String? username;
+
+  User toUserDomain() => User(
```

Comment:
minor: `toDomain()` would suffice
```suggestion
  User toDomain() => User(
```